### PR TITLE
fix: process.exit not immediately exit

### PR DESCRIPTION
**Pull request by lolBig**
##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by lolBig** [comment]
issue: #373

File: src/modules/iotjs_module_process.c [hunk 1]
```diff
@@ -362,15 +362,9 @@ JS_FUNCTION(Chdir) {
...
-  if (!iotjs_environment_is_exiting(env)) {
-    DJS_CHECK_ARGS(1, number);
-    int exit_code = JS_GET_ARG(0, number);
-
-    iotjs_set_process_exitcode(exit_code);
-    iotjs_environment_go_state_exiting(env);
-  }
+  DJS_CHECK_ARGS(1, number);
+  int exit_code = JS_GET_ARG(0, number);
+  exit(exit_code);
```
Comment (legendecas) [inline src/modules/iotjs_module_process.c]: `process.exit(code)` should emit `process#exit` event AFAIK.

Comment (lolBig) [reply src/modules/iotjs_module_process.c]: The `process#exit` event will trigger but after [mustCall](https://github.com/yodaos-project/ShadowNode/blob/3425c729caaa1e37a947252f017f04fb07e6cab3/test/common/index.js#L26)

**Comment by yorkie** [comment]
Any updates here? @lolBig

File: test/run_pass/test_process_exit_handler.js [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+var common = require('../common');
 
-process.on('exit', function() {
-  process.exit(0);
+var fn = (c) => {
+  assert.strictEqual(0, c);
+};
+
+process.on('exit', function(c) {
+  process.exitCode = 123;
```
Comment (legendecas) [inline test/run_pass/test_process_exit_handler.js]: What's the purpose of this line?

File: test/run_pass/test_process_exit_recrusively.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+var assert = require('assert');
+var code = 0;
+var count = 0;
+
+process.on('exit', function(c) {
```
Comment (legendecas) [inline test/run_pass/test_process_exit_recrusively.js]: I do believe `mustCall` has a signature of `mustCall(fn, exact)` of which the second argument should be the call count expected.

File: test/run_pass/test_process_exit_handler.js (hunk 2 above)
Comment (lolBig) [reply test/run_pass/test_process_exit_handler.js]: nothing , i will remove it

**Pull request merged**